model/atmosphere/dycore/docs/icon4py_sphinx.py

```diff
@@ -0,0 +1,364 @@
+
+        docstrings_list = []
+        for idocstr, docstring in enumerate(docstrings):
+            formatted_docstr = None
+
+            # Get some useful information on the following method call
+            call_string = self.get_next_method_call(source, source.find(docstring) + len(docstring))
+            next_method_info = self.get_method_info(call_string)
+            # and process a scientific documentation docstring
+            formatted_docstr = docstring.splitlines()
+            formatted_docstr = self.format_docstring_block(formatted_docstr, self.docstring_keyword)
+            formatted_docstr = self.add_header(formatted_docstr, next_method_info)
+            formatted_docstr = self.process_scidocstrlines(formatted_docstr, next_method_info)
+            formatted_docstr = self.add_next_method_call(formatted_docstr, call_string)
+
+            if formatted_docstr is not None:
+                if idocstr < len(docstrings)-1: # Add footer
+                    formatted_docstr = self.add_footer(formatted_docstr)
+                # add the processed docstring to the list
+                docstrings_list.append(formatted_docstr)
+
+        return docstrings_list
+    
+    def get_docstring_blocks(self, source, keyword):
+        # which in this implementation are comment blocks
+        comment_blocks = []
+        source_lines = source.splitlines()
+        in_block = False
+        current_block = []
+        for line in source_lines:
+            stripped_line = line.strip()
+            if stripped_line.startswith(f'# {keyword}:'):
+                if in_block:
+                    comment_blocks.append('\n'.join(current_block))
+                    current_block = []
+                in_block = True
+                current_block.append(line)
+            elif in_block:
+                if stripped_line.startswith('#'):
+                    current_block.append(line)
+                else:
+                    in_block = False
+                    comment_blocks.append('\n'.join(current_block))
+                    current_block = []
+        if current_block:
+            comment_blocks.append('\n'.join(current_block))
+        return comment_blocks
+
+    def format_docstring_block(self, docstr_lines, keyword):
+        # Clean up and format
+        if docstr_lines[0].strip().startswith(f'# {keyword}:'):
+            docstr_lines.pop(0) # remove the {keyword} prefix
+        while docstr_lines[0].strip() == '#':
+            # strip empty lines from the beginning
+            docstr_lines.pop(0)
+        # strip leading and trailing whitespace of every line (maintain indentation)
+        # as well as comment character and space (indent+2)
+        indent = len(docstr_lines[0]) - len(docstr_lines[0].lstrip(' '))
+        docstr_lines = [line[min(indent+2,len(line)):].rstrip() for line in docstr_lines]
+        # strip empty lines from the end
+        while docstr_lines[-1] == '':
+            docstr_lines.pop(-1)
```

Comment:
Use `textwrap.dedent` from the standard library (https://devdocs.io/python~3.13/library/textwrap#textwrap.dedent)

Reply:
it only works with one string, not a list of strings, which would mean joining them again just for this?